plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AbstractFilesTable.java

```diff
@@ -117,19 +121,30 @@ private List<Page> buildPages()
         PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
         Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
         TableScan tableScan = buildTableScan();
-
+        try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
+            fileScanTasks.forEach(fileScanTask -> {
+                DataTask dataTask = (DataTask) fileScanTask;
+                try (CloseableIterable<StructLike> dataRows = dataTask.rows()) {
+                    dataRows.forEach(dataTaskRow -> addRow(pagesBuilder, idToTypeMapping, (DataFile) dataTaskRow));
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            });
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
         tableScan.planFiles().forEach(fileScanTask -> {
             DataTask dataTask = (DataTask) fileScanTask;
-            dataTask.rows().forEach(dataTaskRow -> addRow(pagesBuilder, idToTypeMapping, dataTaskRow));
+            dataTask.rows().forEach(dataTaskRow -> addRow(pagesBuilder, idToTypeMapping, (DataFile) dataTaskRow));
         });
 
         return pagesBuilder.build();
     }
 
-    private static void addRow(PageListBuilder pagesBuilder, Map<Integer, Type> idToTypeMapping, StructLike dataTaskRow)
+    private static void addRow(PageListBuilder pagesBuilder, Map<Integer, Type> idToTypeMapping, DataFile dataFile)
```

Comment:
This change is not relevant for this commit. Please split to a different commit.

Reply:
added a separate commit.